src/main/java/com/gzs/daos/inmemory/LanguageDaoInMemoryImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.gzs.daos.inmemory;
+
+import com.gzs.daos.LanguageDao;
+import com.gzs.data.GenerateTestData;
+import com.gzs.model.Language;
+
+import java.util.List;
+import java.util.Map;
+
+public class LanguageDaoInMemoryImpl extends InMemoryDao<Language> implements LanguageDao {
```

Comment:
@DespotOfMorea remove test data creation from InMemory DAOs. They don't need to know about it. This way whenever you create InMemoryDao, you'll get that particular test data in it. What someone wants to insert their own test data, or no data at all? This way it is not possible.

It is OK that GenerateTestData exist, but it InMemoryDao should be independent from it, and it should be used manually if needed.

Reply:
@celicni Added setters in GenerateTestData.class and blocked createTestData() in constructor. Now it`s called in RestMethods,if configuration is set for InMemory.

Maybe,split current GenerateTestData.class.Create InMemoryCache.class with maps for Languages,Terms and Translations (with getters and setters) and refer InMemoryImplementation on it.Actual data creation can be moved elsewhere (to new class) and can be called when needed.

Reply:
@DespotOfMorea I think that it is still too inflexible. If I choose InMemory, this particular set of test entities will be created and I can't do anything about it. I might want to use some other test data provider, on none at all. Therefore GenerateTestData should be completely independent from actual Dao layer (Dao layer shouldn't know anything about it). If I want to fill the database with test data I should create instance of GenerateTestData and explicitly call method which fills the database.

Maybe you can expose a rest call which does this so user can invoke rest and initialize test data.

Reply:
@celicni Removed GenerateTestData.class from InMemoryDaos,now they work with empty Map,created in constructor.InMemoryDao.class now have setter for that map,so working data can be added,if needed.

In RestMethods,if data type is set for InMemory,working maps for Term and Translation implementation are filled with data created in GenerateTestData.class.